pallets/parachain-staking/src/lib.rs

```diff
@@ -171,6 +175,9 @@ pub mod pallet {
 		/// Handler to distribute a collator's reward.
 		/// To use the default implementation of minting rewards, specify the type `()`.
 		type PayoutCollatorReward: PayoutCollatorReward<Self>;
+		/// Handler to mark collators as offline.
+		/// To use the default implementation, specify the type `()`.
+		type MarkOfflineCallback: MarkOfflineCallback<Self>;
```

Comment:
```suggestion
		type OnInactiveCollator: OnInactiveCollator<Self>;
```

The goal of the callback is not to mark a collator offline, but to notify the runtime that a collator is inactive, and let the runtime decide what to do ( (which may include marking the collator offline, but not necessarily and not only).